pkg/controllers/updaterun/initialization.go

```diff
@@ -250,6 +250,12 @@ func (r *Reconciler) generateStagesByStrategy(
 		// other err can be retried.
 		return controller.NewAPIServerError(true, err)
 	}
+	// Remove waitTime from the after stage tasks of type Approval.
+	if err := r.removeWaitTimeFromAfterStageTasks(ctx, &updateStrategy); err != nil {
```

Comment:
maybe consider using a mutation webhook to achieve this as this does not guarantee to correct the all the strategy (orphaned one).

Reply:
or we can keep this in until we know there are no more malformated strategy

Reply:
Even with a mutation webhook we won't be able to fix orphaned CSUS?, since it's never being updated and it's only used as an information source